service/src/main/java/uk/gov/hmcts/reform/fpl/service/DocmosisCoverDocumentsService.java

```diff
@@ -17,22 +16,23 @@
 @RequiredArgsConstructor(onConstructor = @__(@Autowired))
 public class DocmosisCoverDocumentsService {
     private final DocmosisDocumentGeneratorService docmosisDocumentGeneratorService;
+    private final ObjectMapper mapper;
 
     public DocmosisDocument createCoverDocuments(String familyManCaseNumber,
                                                  Long caseNumber,
                                                  Representative representative) {
-        Map<String, Object> templateData = buildCoverDocumentsData(familyManCaseNumber, caseNumber, representative);
-        return docmosisDocumentGeneratorService.generateDocmosisDocument(templateData, COVER_DOCS);
+        DocmosisCoverDoc coverDocData = buildCoverDocumentsData(familyManCaseNumber, caseNumber, representative);
+        return docmosisDocumentGeneratorService.generateDocmosisDocument(coverDocData.toMap(mapper), COVER_DOCS);
     }
 
-    Map<String, Object> buildCoverDocumentsData(String familyManCaseNumber,
+    DocmosisCoverDoc buildCoverDocumentsData(String familyManCaseNumber,
                                                 Long caseNumber,
                                                 Representative representative) {
-        return ImmutableMap.<String, Object>builder()
-            .put("familyManCaseNumber", defaultIfNull(familyManCaseNumber, ""))
-            .put("ccdCaseNumber", formatCCDCaseNumber(caseNumber))
-            .put("representativeName", representative.getFullName())
-            .put("representativeAddress", representative.getAddress().getAddressAsString("\n"))
-            .build();
+        DocmosisCoverDoc.Builder coverDocBuilder = DocmosisCoverDoc.builder();
```

Comment:
could we do 

 return DocmosisCoverDoc.builder().familyManCaseNumber(defaultIfNull(familyManCaseNumber, ""))
                            .ccdCaseNumber(formatCCDCaseNumber(caseNumber))
                            .representativeName(representative.getFullName())
                            .representativeAddress(representative.getAddress().getAddressAsString("\n"))
                            .build();

and remove the line DocmosisCoverDoc.Builder coverDocBuilder = DocmosisCoverDoc.builder(); 

?

Reply:
👍updated it